extensions/resource-deployment/src/services/notebookService.ts

```diff
@@ -79,7 +79,9 @@ export class NotebookService implements INotebookService {
 	 * @param content the notebook content
 	 */
 	async launchNotebookWithContent(title: string, content: string): Promise<azdata.nb.NotebookEditor> {
-		const uri: vscode.Uri = vscode.Uri.parse(`untitled:${title}`);
+		const ext = path.extname(title) ?? 'ipynb';
+		title = path.basename(title, ext);
+		const uri: vscode.Uri = vscode.Uri.parse(`untitled:${title}-${getDateTimeString()}.${ext}`);
```

Comment:
I'm also not sure about using the datetime string in the title - that's pretty long for just needing to be a uniquifier. For notebooks we already do this same basic thing but just use a single digit which seems cleaner https://github.com/microsoft/azuredatastudio/blob/master/extensions/notebook/src/extension.ts#L165